src/stdlib/io.rs

```diff
@@ -69,4 +88,831 @@ pub fn load_io<'gc>(ctx: Context<'gc>) {
+                        Value::UserData(file) => {
+                            if let Ok(file) = file.downcast_static::<IoFile>() {
+                                IO_STATE.with(|state| state.replace_output(file.clone()));
+                                Ok(SequencePoll::Return)
+                            } else {
+                                Err("bad argument #1 to 'input' (file expected)"
+                                    .into_value(ctx)
+                                    .into())
+                            }
+                        }
+                        _ => Err("bad argument #1 to 'input' (file expected)"
+                            .into_value(ctx)
+                            .into()),
+                    }
+                }
+            }
+
+            if stack.is_empty() {
+                stack.replace(
+                    ctx,
+                    UserData::new_static(&ctx, IO_STATE.with(|state| state.output())),
+                );
+                return Ok(CallbackReturn::Return);
+            }
+
+            let value = stack.consume::<Value>(ctx)?;
+
+            match value {
+                Value::String(filename) => {
+                    let io: Table = ctx.globals().get(ctx, "io")?;
+                    let open: Callback = io.get(ctx, "open")?;
+
+                    stack.replace(ctx, (filename, "w"));
+                    return Ok(CallbackReturn::Call {
+                        function: open.into(),
+                        then: Some(BoxSequence::new(&ctx, OutputOpenThen)),
+                    });
+                }
+                Value::UserData(file) => {
+                    if let Ok(file) = file.downcast_static::<IoFile>() {
+                        IO_STATE.with(|state| state.replace_output(file.clone()));
+
+                        stack.replace(ctx, UserData::new_static(&ctx, file.clone()));
+                        Ok(CallbackReturn::Return)
+                    } else {
+                        Err("expected `file`".into_value(ctx).into())
+                    }
+                }
+                _ => Err("expected `string` or `file`".into_value(ctx).into()),
+            }
+        }),
+    );
+
+    io.set_field(
+        ctx,
+        "close",
+        Callback::from_fn(&ctx, |ctx, _, mut stack| {
+            if stack.is_empty() {
+                let output = IO_STATE.with(|state| state.output());
+
+                if !output.is_std() {
+                    if let Err(err) = output.close() {
+                        stack.replace(
+                            ctx,
+                            (Value::Nil, err.to_string(), err.raw_os_error().unwrap_or(0)),
+                        );
+                        return Ok(CallbackReturn::Return);
+                    }
+                    IO_STATE.with(|state| state.replace_output(IoFile::stdout()));
+                    stack.replace(ctx, true);
+                    return Ok(CallbackReturn::Return);
```

Comment:
This could be removed, as it's the same as the code that will run after the if statement ends.